Scripts/Python/BaronCityOffice.py

```diff
@@ -84,4 +84,18 @@ def Load(self):
         
     def OnNotify(self,state,id,events):
         pass
-        
\ No newline at end of file
+
+    def OnServerInitComplete(self):
+        dnitime = PtGetDniTime()
+        month = time.strftime('%m', time.gmtime(dnitime))
+        day = time.strftime('%d', time.gmtime(dnitime))
+        agevault = ptAgeVault()
+        if agevault:
+            ageSDL = agevault.getAgeSDL()
+            if ageSDL:
+                if int(month) == 12 and int(day) > 13 and ageSDL.findVar('bcoChristmasVis').getBool() == 0:
+                    ageSDL.findVar('bcoChristmasVis').setBool(1)
+                    agevault.updateAgeSDL(ageSDL)
+                elif ageSDL.findVar('bcoChristmasVis').getBool() == 1:
+                    ageSDL.findVar('bcoChristmasVis').setBool(0)
+                    agevault.updateAgeSDL(ageSDL)
```

Comment:
This makes the code more Pythonic and fixes a logic error by which the Christmas deco will be toggled off on any subsequent link-in. Prefer `""` for strings and `''` for single characters.

```suggestion
        st = time.gmtime(PtGetDniTime())
        if agevault := ptAgeVault():
            if ageSDL := agevault.getAgeSDL():
                if st.tm_mon == 12 and st.tm_mday > 13 and not ageSDL.findVar("bcoChristmasVis").getBool():
                    ageSDL.findVar("bcoChristmasVis").setBool(True)
                    agevault.updateAgeSDL(ageSDL)
                elif not (st.tm_mon == 12 and st.tm_mday > 13) and ageSDL.findVar("bcoChristmasVis").getBool():
                    ageSDL.findVar('bcoChristmasVis').setBool(False)
                    agevault.updateAgeSDL(ageSDL)
```

You'll probably want to propagate these kinds of changes on down...

Reply:
Changes made, please check the code again